src/libraries/Common/src/Interop/Browser/Interop.Runtime.Bridge.cs

```diff
@@ -0,0 +1,381 @@
+        }
+
+        internal static void SetupJSContinuation(Task task, JSObject continuationObj)
+        {
+            if (task.IsCompleted)
+                Complete();
+            else
+                task.GetAwaiter().OnCompleted(Complete);
+
+            void Complete()
+            {
+                try
+                {
+                    if (task.Exception == null)
+                    {
+                        object? result;
+                        Type task_type = task.GetType();
+                        if (task_type == typeof(Task))
+                        {
+                            result = Array.Empty<object>();
+                        }
+                        else
+                        {
+                            result = task_type.GetMethod("get_Result")?.Invoke(task, Array.Empty<object>());
+                        }
+                        continuationObj.Invoke("resolve", result);
+                    }
+                    else
+                    {
+                        continuationObj.Invoke("reject", task.Exception.ToString());
+                    }
+                }
+                catch (Exception e)
+                {
+                    continuationObj.Invoke("reject", e.ToString());
+                }
+                finally
+                {
+                    continuationObj.Dispose();
+                    FreeObject(task);
+                }
+            }
+        }
+
+        internal static string ObjectToString(object o)
+        {
+            return o.ToString() ?? string.Empty;
+        }
+
+        internal static double GetDateValue(object dtv)
+        {
+            if (dtv == null)
+                throw new ArgumentNullException(nameof(dtv), "Value can not be null");
+            if (!(dtv is DateTime dt))
+                throw new InvalidCastException($"Unable to cast object of type {dtv.GetType()} to type DateTime.");
+            if (dt.Kind == DateTimeKind.Local)
+                dt = dt.ToUniversalTime();
+            else if (dt.Kind == DateTimeKind.Unspecified)
+                dt = new DateTime(dt.Ticks, DateTimeKind.Utc);
+            return new DateTimeOffset(dt).ToUnixTimeMilliseconds();
+        }
+
+        internal static DateTime CreateDateTime(double ticks)
+        {
+            var unixTime = DateTimeOffset.FromUnixTimeMilliseconds((long)ticks);
+            return unixTime.DateTime;
+        }
+
+        internal static Uri CreateUri(string uri)
+        {
+            return new Uri(uri);
+        }
+
+        //
+        // Can be called by the app to stop profiling
+        //
+        [MethodImplAttribute(MethodImplOptions.NoInlining)]
+        public static void StopProfile()
+        {
+        }
+
+        // Called by the AOT profiler to save profile data into Module.aot_profile_data
+        internal static unsafe void DumpAotProfileData(ref byte buf, int len, string s)
+        {
+            var arr = new byte[len];
```

Comment:
Where is this used?

Reply:
This is called from https://github.com/dotnet/runtime/pull/35566 from the lib_mono.js included library code.  mono_wasm_init_aot_profiler:

All of the methods in the Bridge code module are those methods that are called from the wasm runtime build glue code.  Not sure if this will still be implemented though or how.

Maybe @marek-safar can shed some light on this.

Reply:
We can remove it from this initial PR

Reply:
Addressed in recent commit

Reply:
To be clear, though, I was referring to `arr`.